app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,98 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Create a conversion host and enable it. This operation will run as an
+    # MiqTask.
+    #
+    # Both the 'resource_type' and 'resource_id' are mandatory arguments,
+    # and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "resource_type": "Host",
+    #   "resource_id": "7"
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(type, id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+
+      resource_type = data['resource_type']
+      collection_type = resource_type.include?('Host') ? 'hosts' : 'vms'
```

Comment:
do we know what will UI send to us? The base class? or specific class e.g. with specific provider?
Also, does the `collection_class` handle only a subset of classes? or any class?

(I am trying to see if this can be generic as well)

Reply:
@jameswnl My expectation is that it will send the full class name, e.g. ManageIQ::Providers::Redhat::InfraManager::Host, but this handles it either way.

Pretty sure collection_class handles any class that's had a collection defined for it. Not entirely sure what you mean here by generic.

Reply:
Automatically assuming it's `vms` if it is not `hosts` seems potentially fragile to me.

Reply:
we should probably let it fail if we can't recognize the incoming type. we can't just assume it

Reply:
It will automatically fail once the updated validation is merged.

Reply:
Hm, for a more generic approach, how about `resource_type.split('::').last.downcase.pluralize`

Reply:
Or we could approach it from the other direction and do `resource_type.classify.constantize.table_name`.

Reply:
@jameswnl Ok, updated.